llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -10296,10 +10296,10 @@ class BoUpSLP::ShuffleCostEstimator : public BaseShuffleAnalysis {
             Idx = EMask[Idx];
         }
         CommonVF = E->Scalars.size();
-      } else if (std::optional<unsigned> Factor = E->getInterleaveFactor();
```

Comment:
Remove the useless std::optional as a pre-commit

Reply:
Done